docs/UserGuide.md

```diff
@@ -5,44 +5,111 @@ title: User Guide
 
 ![gitGud](images/gitgud.png)
 
-gitGud is a **desktop application for managing your friends' gaming information**. It uses a gamer-themed **Graphical User Interface 
-(GUI) combined with an optimized Command Line Interface (CLI)** to give you a smooth and efficient experience. 
+* Table of Contents
+  {:toc}
```

Comment:
Consider removing the space before `{:toc}` or the toc will not be automatically generated. (This was the reason our TOC did not auto-generate the last time)